pytorch_lightning/utilities/distributed.py

```diff
@@ -88,22 +96,60 @@ def gather_all_tensors_if_available(result: Union[torch.Tensor], group: Optional
 
     """
     if torch.distributed.is_available() and torch.distributed.is_initialized():
-        if group is None:
-            group = torch.distributed.group.WORLD
+        result = gather_ddp(result, group)
+    if HOROVOD_AVAILABLE and hvd.is_initialized():
+        return gather_horovod(result, group=group)
```

Comment:
@ananyahjha93 and @tgaddair please use accelerators. no more horovod or other backend code floating around.

```python
self.accelerator.gather(...)
```

Reply:
Hey @williamFalcon, I would love to use acceletators here, but the issue is that because these `EvalResult` and `Metric` instances are created and executed in user code, there's currently no clean way to inject the accelerator state from the `Trainer`.  That's why both Horovod and DDP are looking at global state here.

Happy to refactor to use accelerators, but we may need to jam on how best to achieve this with the existing API.  For example, you could have a factory: `trainer.CreateEvalResult`, but that's not super elegant.